script.js

```diff
@@ -219,19 +233,35 @@ function validateEmail(input, requiredMsg, invalidMsg) {
 
 const form = document.querySelector('#signup');
 
-const NAME_REQUIRED = 'Please enter your name';
+const NAME_REQUIRED = 'Please enter your full name';
 const EMAIL_REQUIRED = 'Please enter your email';
 const EMAIL_INVALID = 'Please enter a correct email address format';
+const FIRST_NAME_REQUIRED = 'Please enter your first name';
+const LAST_NAME_REQUIRED = 'Please enter your last name';
 
 form.addEventListener('submit', (event) => {
   event.preventDefault();
 
-  const nameValid = hasValue(form.elements.name, NAME_REQUIRED);
-  const emailValid = validateEmail(form.elements.email, EMAIL_REQUIRED, EMAIL_INVALID);
-  const firstNameValid = hasValue(form.elements['first-name'], NAME_REQUIRED);
-  const lastNameValid = hasValue(form.elements['last-name'], NAME_REQUIRED);
+  const fullName = fullNameInput.value.trim();
+  const email = emailInput.value.trim();
+  const firstName = firstNameInput.value.trim();
+  const lastName = lastNameInput.value.trim();
+
+  const nameValid = hasValue(fullNameInput, NAME_REQUIRED);
+  const emailValid = validateEmail(emailInput, EMAIL_REQUIRED, EMAIL_INVALID);
+  const firstNameValid = hasValue(firstNameInput, FIRST_NAME_REQUIRED);
+  const lastNameValid = hasValue(lastNameInput, LAST_NAME_REQUIRED);
 
   if (((firstNameValid && lastNameValid) || (nameValid)) && emailValid) {
     form.submit();
   }
+
+  const formInfo = {
+    fullName,
+    email,
+    firstName,
+    lastName,
+  };
+
+  localStorage.setItem('formInfo', JSON.stringify(formInfo));
 });
```

Comment:
- [ ] I noticed that your details only persist on submit of the form. Please note that the project requires the values of the input fields to be stored in the local storage on change of each input field. Kindly fix that, please. You can use the 'input' event on the form.